File: solutions/kotlin/sum-of-multiples/1/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {
    fun sum(factors: Set<Int>, limit: Int): Int {
        return (1 until limit)
            .filter { number -> 
                factors.filter { it > 0 }.any { factor -> number % factor == 0 } 
            }
            .sum()
    }
}

File: solutions/julia/sum-of-multiples/1/sum-of-multiples.jl
function sum_of_multiples(limit, factors)
    multiples = Set{Int}()
    
    for f in factors
        if f > 0
            for i in f:f:(limit - 1)
                push!(multiples, i)
            end
        end
    end
    
    return sum(multiples; init=0)
end
